pkg/bgpv1/manager/reconcilerv2/crd_status_test.go

```diff
@@ -5,24 +5,319 @@ package reconcilerv2
+					Error:    "error 00",
+				},
+				{
+					Instance: "bgp-instance-0",
+					ErrorID:  1,
+					Error:    "error 01",
+				},
+				{
+					Instance: "bgp-instance-1",
+					ErrorID:  0,
+					Error:    "error 10",
+				},
+			},
+			initNodeConfig: &v2alpha1.CiliumBGPNodeConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node0",
+				},
+			},
+			expectedNodeConfig: &v2alpha1.CiliumBGPNodeConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node0",
+				},
+				Spec: v2alpha1.CiliumBGPNodeSpec{},
+				Status: v2alpha1.CiliumBGPNodeStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:   v2alpha1.BGPInstanceConditionReconcileError,
+							Status: metav1.ConditionTrue,
+							Reason: "BGPReconcileError",
+							Message: "ReconcileError{Instance: bgp-instance-0, ErrorID: 0, Error: error 00}\n" +
+								"ReconcileError{Instance: bgp-instance-0, ErrorID: 1, Error: error 01}\n" +
+								"ReconcileError{Instance: bgp-instance-1, ErrorID: 0, Error: error 10}\n",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "modify previous error conditions",
+			statedbData: []*tables.ReconcileError{
+				{
+					Instance: "bgp-instance-0",
+					ErrorID:  0,
+					Error:    "error 00",
+				},
+			},
+			initNodeConfig: &v2alpha1.CiliumBGPNodeConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node0",
+				},
+				Spec: v2alpha1.CiliumBGPNodeSpec{},
+				Status: v2alpha1.CiliumBGPNodeStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:   v2alpha1.BGPInstanceConditionReconcileError,
+							Status: metav1.ConditionTrue,
+							Reason: "BGPReconcileError",
+							Message: "ReconcileError{Instance: bgp-instance-0, ErrorID: 0, Error: error 00}\n" +
+								"ReconcileError{Instance: bgp-instance-0, ErrorID: 1, Error: error 01}\n" +
+								"ReconcileError{Instance: bgp-instance-1, ErrorID: 0, Error: error 10}\n",
+						},
+					},
+				},
+			},
+			expectedNodeConfig: &v2alpha1.CiliumBGPNodeConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node0",
+				},
+				Status: v2alpha1.CiliumBGPNodeStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:    v2alpha1.BGPInstanceConditionReconcileError,
+							Status:  metav1.ConditionTrue,
+							Reason:  "BGPReconcileError",
+							Message: "ReconcileError{Instance: bgp-instance-0, ErrorID: 0, Error: error 00}\n",
+						},
+					},
+				},
+			},
+		},
+		{
+			name:        "delete previous error conditions",
+			statedbData: []*tables.ReconcileError{},
+			initNodeConfig: &v2alpha1.CiliumBGPNodeConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node0",
+				},
+				Spec: v2alpha1.CiliumBGPNodeSpec{},
+				Status: v2alpha1.CiliumBGPNodeStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:   v2alpha1.BGPInstanceConditionReconcileError,
+							Status: metav1.ConditionTrue,
+							Reason: "BGPReconcileError",
+							Message: "ReconcileError{Instance: bgp-instance-0, ErrorID: 0, Error: error 00}\n" +
```

Comment:
Wondering whether we should apply some nicer printing here - I guess the main consumers of this output are human operators, right? Maybe we could omit the ErrorID (which seems to be internal), and print something like:
```
Instance-name-0: Error1.
Instance-name-0: Error2.
Instance-name-1: Error3.
```